### PR TITLE
Parameter handling

### PR DESCRIPTION
**Pull request by martwo**
Merging the parameter_handling branch with the master branch.

### CODE REVIEW

File: skyllh/analyses/i3/publicdata_ps/utils.py [hunk 1]
```diff
@@ -253,3 +310,29 @@ def create_energy_cut_spline(ds, exp_data, spl_smooth):
...
+    src_array = tdm.get_data('src_array')
+    src_ra = np.take(src_array['ra'], src_idxs)
+    src_dec = np.take(src_array['dec'], src_idxs)
+
+    psi = angular_separation(
+        ra1=ra,
+        dec1=dec,
+        ra2=src_ra,
+        dec2=src_dec,
+        psi_floor=10**-5.95442953)
```
Comment (tomaskontrimas) [inline skyllh/analyses/i3/publicdata_ps/utils.py:336]: The given `psi_floor` value here is only needed for the correct KDE PDFs evaluation, due to the way how photosplines were built. I think that for analyses which are using Gaussian spatial term one could set this to 0 by default. One way to make it more generic would be to provide `get_tdm_field_func_psi` function `with psi_floor=0` kwarg, which would set the `psi_floor` and return `tdm_field_func_psi`.

**Review by tomaskontrimas — COMMENTED** [review]
Review of 40/145 files viewed

**Review by tomaskontrimas — COMMENTED** [review]
110/145 files viewed

File: skyllh/core/utils/multidimgridpdf.py
Comment (tomaskontrimas) [inline skyllh/core/utils/multidimgridpdf.py:1]: The `kind` argument is no longer used. Should be removed from kwargs/docstring parameters and description

**Review by tomaskontrimas — COMMENTED** [review]
Final comments

File: doc/sphinx/concepts/flux_definition.ipynb
[the remark this answers is not in the document]
Comment (martwo) [reply doc/sphinx/concepts/flux_definition.ipynb:1]: I fixed that

File: doc/sphinx/concepts/pdf.ipynb
[the remark this answers is not in the document]
Comment (martwo) [reply doc/sphinx/concepts/pdf.ipynb:1]: Fixed

File: doc/sphinx/tutorials/publicdata_ps_timedep.ipynb
[the remark this answers is not in the document]
Comment (martwo) [reply doc/sphinx/tutorials/publicdata_ps_timedep.ipynb:1]: Fixed

File: doc/user_manual.tex [hunk 2]
```diff
@@ -730,40 +759,44 @@ \subsection{Gradients of the Multi-Dataset Log-Likelihood Ratio}
...
-quantities: right-ascension, declination, and observation time, i.e.
-$\xs = (\alpha_{\mathrm{s}},\delta_{\mathrm{s}},t_{\mathrm{obs}})$.
 
-The time-dependent effective area $A_{\mathrm{eff}_j}(E,t|\xsk)$ must account
+% In the most-general case, the source position $\xs$ consists of celestrial
+% coordinates right-ascension and declination, which might be a function of the
+% observation time $t_{\mathrm{obs}}$ for very long data taking periods in the
+% order of several tens of years, i.e. due to the movement of the solar system
+% w.r.t. the source:
+% $\xs = \{\alpha_{\mathrm{s}}(t_{\mathrm{obs}}),\delta_{\mathrm{s}}(t_{\mathrm{obs}})\}$.
```
[the remark this answers is not in the document]
Comment (martwo) [reply doc/user_manual.tex]: I uncommented these lines.

File: doc/user_manual.tex [hunk 3]
```diff
@@ -786,99 +819,30 @@ \subsection{Effective Area}
...
-% \subsection{Time Dependent Point-Source Flare}
-%
-% The \class{TimeDepPSFlareLHModel} class provides the likelihood model for searching for a point source with unknown time-dependence.
-% The search is based on the formulism described in \cite{TimeDepPSSearchMethods2010}.
-%
-% The model utilizes a two-component likelihood function with signal and background events.
+This section describes the implemented log-likelihood models.
+\cite{TimeDepPSSearchMethods2010}
+
+
```
[the remark this answers is not in the document]
Comment (martwo) [reply doc/user_manual.tex:846]: I created an issue

File: skyllh/analyses/i3/publicdata_ps/utils.py (hunk 1 above)
Comment (chiarabellenghi) [reply skyllh/analyses/i3/publicdata_ps/utils.py:336]: In SkyLLH as it is right now, the `psi_func` to compute the angular distance between the events and the source is in the analysis scripts, e.g. `time_integrated_ps.py` and `time_dependent_ps.py`.

In that function, no psi_floor is being applied: https://github.com/icecube/skyllh/blob/009c6a7cbd0417d13c9cbcf17bbca4e0fa3feb9f/skyllh/analyses/i3/publicdata_ps/time_integrated_ps.py#L90-L117

**Review by tomaskontrimas — APPROVED** [review]
💯

**Pull request merged**